File: 8-kyu/check-same-case/c/solution.c
#include <ctype.h>
#include <string.h>

int same_case (char a, char b)
{
    if (!isalpha(a) || !isalpha(b))
    {
        return -1;
    }
    else if (islower(a) == islower(b) || isupper(a) == isupper(b))
    {
        return 1;
    }
    return 0;
}

File: 8-kyu/check-same-case/javascript/solution.js
const sameCase = (a, b) => {
  const islower = ch => ch.toLowerCase() === ch;
  const isupper = ch => ch.toUpperCase() === ch;
  const isalpha = ch => /^[a-z]$/i.test(ch);
  if (!isalpha(a) || !isalpha(b)) {
    return -1;
  }
  return islower(a) === islower(b) || isupper(a) === isupper(b) ? 1 : 0;
};
